torch/testing/_internal/common_distributed.py

```diff
@@ -842,6 +852,66 @@ def _check_return_codes(self, elapsed_time) -> None:
     def is_master(self) -> bool:
         return self.rank == 0
 
+# Utility base class for DDP based Multi Process Test cases
+# This abstracts the PG creation and deletion, the backends are selected based
+# on device type. The tests functions can be instantiated per device type using
+# common_device_type.instantiate_device_type_tests
+# other backends can add entry in backend() function
+class DistributedTestBase(MultiProcessTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self._spawn_processes()
+
+    def tearDown(self):
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    def backend(self, device) -> str:
```

Comment:
The device-backend logic should probably live in distributed_c10d.py somewhere. I'm not sure the best way to do it, but it would seem reasonable to have a helper that is usable not only in tests

Reply:
Sure, I think in general we have lot of scope of harmonizing the APIs for different devices in torch.distributed , can I  add that as part of a follow up PR?